core/src/test/java/org/bitcoinj/core/ECKeyTest.java

```diff
@@ -72,15 +70,18 @@ public void setUp() throws Exception {
     public void sValue() throws Exception {
         // Check that we never generate an S value that is larger than half the curve order. This avoids a malleability
```

Comment:
For both the old and the new code I miss where the actual "check" is executed. Do you know?

Reply:
The checks are performed _after_ the async calculations complete and the results of the completed futures are assembled in a list. This starts at line 85 in the "after" code of this PR:

```
for (ECKey.ECDSASignature signature : sigs) {
     assertTrue(signature.isCanonical());
}
```

Reply:
You're right, thanks for pointing this out!